formats/utils/color-factory.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable id-length */
+const tinycolor = require('tinycolor2');
+const {hexToHsluv, hsluvToRgb} = require('hsluv');
+
+const colorFactory = (colors, roleVariants, colorScheme) => {
+  return Object.assign(
+    {},
+    ...Object.entries(colors).map(([role, hex]) => {
+      const base = hexToHsluv(hex);
+      const variants = roleVariants[role] || [];
+      return {
+        ...variants.reduce((accumulator, {name, ...settings}) => {
+          const {
+            hue = base[0],
+            saturation = base[1],
+            lightness = base[2],
+            alpha = 1,
+          } = settings[colorScheme];
+
+          const resolve = (value, baseToResolve) => {
+            return typeof value === 'number' ? value : value(baseToResolve);
+          };
+
+          const rgbColor = hsluvToRgb([
```

Comment:
Currently getting some weird decimal values out of this `hsluvToRgb` function, so all the colors being output from this factory are broken. Will need to explore more.

Reply:
Looks like `hsluv` had a `1.0.0` release just a couple days ago, which means polaris react and polaris tokens are no longer on the same version.

That explains why I needed to update the function calls, since the API has changed completely.

Will need to target `"hsluv": "^0.0.3",` instead.

Reply:
That doesn't seem to be it either. The `hsluvToRgb` function appears to be straight up broken.